helper/parseItems.js

```diff
@@ -1,120 +1,143 @@
-        fishing_bag: INVENTORY?.bag_contents?.fishing_bag?.data ?? '',
-        potion_bag: INVENTORY?.bag_contents?.potion_bag?.data ?? '',
-        sacks_bag: INVENTORY?.bag_contents?.sacks_bag?.data ?? '',
-        candy_inventory: SHARED_INVENTORY?.candy_inventory_contents?.data ?? '',
-        carnival_mask_inventory: SHARED_INVENTORY?.carnival_mask_inventory_contents?.data ?? '',
-        quiver: INVENTORY?.bag_contents?.quiver?.data ?? '',
+const parseItems = async (profileData, museumData, options = { removeEmptyItems: true, combineStorage: true, additionalInventories: null }) => {
+    // Prepare data
+    const INVENTORY = profileData.inventory ?? {};
+    const SHARED_INVENTORY = profileData.shared_inventory ?? {};
 
+    // Setup inventory data
+    const rawData = {
+        armor: INVENTORY.inv_armor?.data ?? '',
+        equipment: INVENTORY.equipment_contents?.data ?? '',
+        wardrobe: INVENTORY.wardrobe_contents?.data ?? '',
+        inventory: INVENTORY.inv_contents?.data ?? '',
+        enderchest: INVENTORY.ender_chest_contents?.data ?? '',
+        accessories: INVENTORY.bag_contents?.talisman_bag?.data ?? '',
+        personal_vault: INVENTORY.personal_vault_contents?.data ?? '',
+        fishing_bag: INVENTORY.bag_contents?.fishing_bag?.data ?? '',
+        potion_bag: INVENTORY.bag_contents?.potion_bag?.data ?? '',
+        sacks_bag: INVENTORY.bag_contents?.sacks_bag?.data ?? '',
+        candy_inventory: SHARED_INVENTORY.candy_inventory_contents?.data ?? '',
+        carnival_mask_inventory: SHARED_INVENTORY.carnival_mask_inventory_contents?.data ?? '',
+        quiver: INVENTORY.bag_contents?.quiver?.data ?? '',
         ...Object.fromEntries([
             ...Object.entries(INVENTORY?.backpack_contents ?? {}).map(([key, value]) => [`storage_${key}`, value.data ?? '']),
             ...Object.entries(INVENTORY?.backpack_icons ?? {}).map(([key, value]) => [`storage_icon_${key}`, value.data ?? '']),
         ]),
+        ...(options.additionalInventories ?? {}),
     };
 
-    const entries = Object.entries(outputPromises);
-    const decodedItems = await decodeItems(entries.map(([_, value]) => value));
+    // Decode data
+    const decodedItems = await decodeItems(Object.values(rawData));
 
-    const items = entries.reduce((acc, [key, _], idx) => {
-        if (!decodedItems[idx]) {
-            acc[key] = [];
-            return acc;
-        }
+    // Prepare return object
+    const items = {};
+    if (options.combineStorage) {
+        items.storage = [];
+    }
 
-        const filteredItems = decodedItems[idx].filter((item) => item && Object.keys(item).length);
-        if (key.includes('storage')) {
-            acc.storage = (acc.storage || []).concat(filteredItems);
-        } else {
-            acc[key] = filteredItems;
+    // Loop through inventories
+    Object.keys(rawData).forEach((key, idx) => {
+        let currentInventoryItems = decodedItems[idx];
```

Comment:
This will not work since the length of `Object.keys(rawData)` and the length of `decodedItems` can differ due to filtering in `decodeItems`.